sktime/forecasting/compose/_pipeline.py

```diff
@@ -1210,29 +1210,43 @@ class ForecastX(BaseForecaster):
     ----------
     forecaster_y : BaseForecaster
         sktime forecaster to use for endogeneous data ``y``
+
     forecaster_X : BaseForecaster
         sktime forecaster to use for exogeneous data ``X``
+
     fh_X : None, ForecastingHorizon, or valid input to construct ForecastingHorizon
         optional, default = None = same as used for ``y`` in any instance.
         valid inputs to construct ``ForecastingHorizon`` are:
         int, list of int, 1D np.ndarray, pandas.Index (see ForecastingHorizon)
+
     behaviour : str, one of "update" or "refit", optional, default = "update"
-        if "update", ``forecaster_X`` is fit to the data batch seen in ``fit``,
-            and updated with any ``X`` seen in calls of ``update``.
-            Forecast added to ``X`` in ``predict`` is obtained from this state.
-        if "refit", then ``forecaster_X`` is fit to ``X`` in ``predict`` only,
-            Forecast added to ``X`` in ``predict`` is obtained from this state.
+
+        * if "update", ``forecaster_X`` is fit to the data batch seen in ``fit``,
+        and updated with any ``X`` seen in calls of ``update``.
+        Forecast added to ``X`` in ``predict`` is obtained from this state.
+
+        * if "refit", then ``forecaster_X`` is fit to ``X`` in ``predict`` only,
+        Forecast added to ``X`` in ``predict`` is obtained from this state.
+
     columns : None, or pandas compatible index iterator (e.g., list of str), optional
         default = None = all columns in ``X`` are used for forecast columns to which
         ``forecaster_X`` is applied.
         If not ``None``, must be a non-empty list of valid column names.
         Note that ``[]`` and ``None`` do not imply the same.
-    fit_behaviour : str, one of "use_actual", "use_forecast", optional,
-        default = "use_actual"
-        if "use_actual", then ``forecaster_y`` uses the actual ``X`` as
-            exogenous features in `fit`
-        if "use_forecast", then ``forecaster_y`` uses the ``X`` predicted by
-            ``forecaster_X`` as exogenous features in ``fit``
+
+    fit_behaviour : str, one of "use_actual" (default), "use_forecast", optional,
+
+        * if "use_actual", then ``forecaster_y`` uses the actual ``X`` as
+        exogenous features in `fit`
+        * if "use_forecast", then ``forecaster_y`` uses the ``X`` predicted by
+        ``forecaster_X`` as exogenous features in ``fit``
+
+    forecaster_X_exogeneous : str, one of "None" (default), or "complement"
+
+        * if "None", then ``forecaster_X`` uses no exogenous data
+        * if "complement", then ``forecaster_X`` uses the complement of the
+        ``columns`` as exogenous data to forecast. This is typically useful
+        if the complement of ``columns`` is known to be available in the future.
```

Comment:
Can we generalise it to list of column names, similar to `columns`? May be the users will like to control what future-known X to use to predict future-unknown X.

As far as I can see, the only code change required will be in the `_get_X_for_fcX` function, where in a new `if` case we need to do `X_for_fcX = X[forecaster_X_exogeneous]` and etc.

Reply:
sure - do you find that easier than chaining `forecaster_X` with `IndexSubset` or the `__getitem__` dunder?

Reply:
added support for this